### PR TITLE
Demo for LERC

### PR DESCRIPTION
**Pull request by anthonywchen**
Hi! Here's the start of my adding the demo code for LERC. 

For now, I've believe I finished adding the API code into `api/allennlp_demo/lerc/` for LERC. I've done the following:
* `api/allennlp_demo/lerc/Dockerfile`: Custom Dockerfile because the model/predictor code lives in my own repo.
* `api/allennlp_demo/lerc/README.md`: Has commands to run `api/allennlp_demo/lerc/test_api.py`. I was able to run this test and get the appropriate output.
*`api/allennlp_demo/nmn/README.md`: I wasn't able to run the test with the commands before, but upon fixing the command, I was able to.
* `api/allennlp_demo/requirements.txt`: I've changed the `allennlp-models` to just point to master. I did this because the branch it points to now doesn't have the LERC model card and evaluate-rc task card. Let me know if I shouldn't point this at master.

I'll upload the changes I have for the UI next.

### CODE REVIEW

**Comment by anthonywchen** [comment]
Tagging @jonborchardt @dirkgr! Thanks!

**Review by dirkgr — APPROVED** [review]
LGTM, but I really want @jonborchardt to take a look too.

**Comment by jonborchardt** [comment]
seems fine to me, adding @codeviking incase there is an issue
i say resolve conflicts, merge, and test!

File: api/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 git+git://github.com/allenai/allennlp.git@v2.1.0
-git+git://github.com/allenai/allennlp-models.git@bbbf456abf6a1f1c1a852af0d3cfd82147f32a89
+git+git://github.com/allenai/allennlp-models.git
```
Comment (codeviking) [inline api/requirements.txt:2]: @dirkgr, @epwalsh or @AkshitaB should sign off on this. I think we run use a very specific AllenNLP Models version for a reason.

Comment (dirkgr) [reply api/requirements.txt:2]: I sign off on this. We sometimes pin the version for specific needs, but should not need to do it in general. The promise of showing those instructions is that it works when you get the latest version and run it.

**Review by codeviking — COMMENTED** [review]
Looks like you're missing the `.skiff` directory. You'll want that too.
Here's an example: https://github.com/allenai/allennlp-demo/tree/master/api/allennlp_demo/bidaf/.skiff

There are docs for `APIEndpoint()` can be found here:
https://github.com/allenai/allennlp-demo/blob/master/api/allennlp_demo/common/.skiff/common.libsonnet#L41

**Review by codeviking — APPROVED** [review]
LGTM.

It looks like there's a formatting error (that's why CI is failing). I think you can fix it by doing this:

```
cd api/
black allennlp_demo
```

**Comment by anthonywchen** [comment]
Thanks Sam. Is this something I should fix or is this something that needs to be fixed on the CI side of things?

**Comment by codeviking** [comment]
Yea, I think so. It's a formatting error in the file `allennlp-demo/allennlp-demo/api/allennlp_demo/lerc/test_api.py`,  which I think was added in this PR.

**Comment by anthonywchen** [comment]
I didn't know about black; cool stuff!

**Comment by anthonywchen** [comment]
Alright, so I also started added the UI code for this new task of evaluating reading comprehension. I think I'm really close to finishing this part. The thing that needs to be fixed is these four lines [here](https://github.com/anthonywchen/allennlp-demo/blob/8c1105a17a20c2979c13858735fe86e4dfbef934/ui/src/demos/evaluate-reading-comprehension/Main.tsx#L34-L37). 

I'm not sure how `Fields` interacts with the input JSON data, but I think I need Fields for `Context`, `Reference` and `Candidate` defined (not sure where `@allenai/tugboat` is defined). Here is what in input JSON data point looks like for this task:

```
{
    "context": "Robin wept in front of Skylar who was sick of seeing her cry.",
    "question": "What will happen to Robin?",
    "reference" : "be scolded",
    "candidate" : "be sad"
}
```

**Comment by jonborchardt** [comment]
> Alright, so I also started added the UI code for this new task of evaluating reading comprehension. I think I'm really close to finishing this part. The last thing that needs to be worked on is these four lines [here](https://github.com/anthonywchen/allennlp-demo/blob/8c1105a17a20c2979c13858735fe86e4dfbef934/ui/src/demos/evaluate-reading-comprehension/Main.tsx#L34-L37).
> 
> I'm not sure how `Fields` interacts with the input JSON data, but I think I need Fields for `Context`, `Reference` and `Candidate` defined (not sure where `@allenai/tugboat` is defined). Here is what in input JSON data point looks like for this task:
> 
> ```
> {
>     "context": "Robin wept in front of Skylar who was sick of seeing her cry.",
>     "question": "What will happen to Robin?",
>     "reference" : "be scolded",
>     "candidate" : "be sad"
> }
> ```

i can add those to tugboat... do you only need `reference` and `candidate` as one line text input?

**Comment by anthonywchen** [comment]
Thank Jon! Yup, `Reference` and `Candidate` will just be a string each.

**Comment by jonborchardt** [comment]
> Thanks Jon! Yup, `Reference` and `Candidate` will just be a string each.

can you convert your api to take 'passage' or 'document' instead of 'context'

**Comment by anthonywchen** [comment]
Yeah I can try to do that. Doing that would either require me to change my Predictor/Model. I think I can overwrite one of the `http.ModelEndpoint` functions to take in a `context` field, then pass it to the Predictor as a `passage`.

**Comment by jonborchardt** [comment]
> Yeah I can try to do that. Doing that would either require me to change my Predictor/Model. I think I can overwrite one of the `http.ModelEndpoint` functions to take in a `context` field, then pass it to the Predictor as a `passage`.

ill add context in tugboat

**Comment by anthonywchen** [comment]
ah ok, sounds good, thanks!

**Comment by jonborchardt** [comment]
ok, update the package.json to have `"tugboat": "^2.0.1"`
and you sho-uld be golden to use
```
<Context />
<Reference />
<Candidate />
```

**Comment by anthonywchen** [comment]
awesome, thanks! I'll give  it a shot now

**Comment by codeviking** [comment]
@anthonywchen @jonborchardt and I are taking off for the day. If you run into any other issues let us know and we'll TAL on Monday.

We'd also love to get your general feedback about the demo building experience after you're done. If you have ideas for making it easier, or specific things that were tricky, we'd love to hear about them.

Thanks and have a great weekend!

**Comment by anthonywchen** [comment]
thanks a lot for your help! I should be able to take things from here :) And I'd be happy to share some thoughts. Happy to send an email or set up a meeting to chat :)

**Comment by codeviking** [comment]
@anthonywchen An email is great. Go ahead and send ideas to reviz@allenai.org.  Thanks!

**Comment by anthonywchen** [comment]
Thanks Sam. I'll send an email soon. I also wanted to give an update and say I think the UI is completed! Please let me know if there's anything I'm missing.

**Comment by dirkgr** [comment]
Is this all ready for reviews/merges then?

**Comment by anthonywchen** [comment]
@dirkgr Yes, it is ready at this point. Thanks!

**Comment by dirkgr** [comment]
I just ran this for the first time. Looks neat!

It shows up under the heading "Compare two sentences". I guess it does that, but since it's labeled "Evaluating Reading Comprehension", I would have expected it under "Question answering". I'm happy with either location, but I wanted to let you know.

**Comment by dirkgr** [comment]
I am looking into why this makes VilBERT fail. That should not be happening.

**Comment by dirkgr** [comment]
And it's live!

**Comment by anthonywchen** [comment]
Yay, thank you (Dirk, Sam, Jon) very much for help! :)

**Pull request merged**